src/main/scala/viper/gobra/reporting/VerifierError.scala

```diff
@@ -181,6 +181,11 @@ case class AssertError(info: Source.Verifier.Info) extends VerificationError {
   override def localMessage: String = "Assert might fail"
 }
 
+case class DeriveError(info: Source.Verifier.Info) extends VerificationError {
+  override def localId: String = "derive_error"
+  override def localMessage: String = "Derive statement failed. The body might not derive a contradiction after assuming the negated expression"
```

Comment:
I think this may lead to noisy error messages, where the "title" message is too long. It may be more consistent with the rest of Gobra to have the `localMessage` here be "Derive statement failed" and then we can introduce an ErrorReason whose message contains the rest of the message above, similarly to how it is done for assertion errors